spec/lib/rx/client_spec.rb

```diff
@@ -16,69 +16,131 @@
+      VCR.use_cassette("rx_client/preferences/#{cassette}") do
+        expect { client.post_preferences(email_address: 'kamyar karshenas@va.gov', rx_flag: false) }
+          .to raise_error(Common::Exceptions::BackendServiceException)
+      end
     end
   end
 
-  describe 'prescriptions' do
-    it 'gets a list of active prescriptions', :vcr do
-      client_response = client.get_active_rxs
-      expect(client_response).to be_a(Common::Collection)
-      expect(client_response.members.first).to be_a(Prescription)
+  shared_examples 'prescriptions' do |caching_enabled|
+    before(:each) do
+      Settings.mhv.rx.collection_caching_enabled = caching_enabled
     end
 
-    it 'gets a list of all prescriptions', :vcr do
-      client_response = client.get_history_rxs
-      expect(client_response).to be_a(Common::Collection)
-      expect(client_response.members.first).to be_a(Prescription)
+    let(:cache_keys) { ["#{client.session.user_id}:getactiverx", "#{client.session.user_id}:gethistoryrx"] }
+
+    it 'gets a list of active prescriptions' do
+      VCR.use_cassette('rx_client/prescriptions/gets_a_list_of_active_prescriptions') do
+        client_response = client.get_active_rxs
+        expect(client_response).to be_a(Common::Collection)
+        expect(client_response.type).to eq(Prescription)
+        expect(client_response.cached?).to eq(caching_enabled)
+
+        if caching_enabled
+          expect(cache_key_for(client_response)).to eq("#{client.session.user_id}:getactiverx")
+        else
+          expect(cache_key_for(client_response)).to eq(nil)
+        end
+      end
     end
 
-    it 'gets a single prescription', :vcr do
-      expect(client.get_rx(13_650_546)).to be_a(Prescription)
+    it 'gets a list of all prescriptions' do
+      VCR.use_cassette('rx_client/prescriptions/gets_a_list_of_all_prescriptions') do
+        client_response = client.get_history_rxs
+        expect(client_response).to be_a(Common::Collection)
+        expect(client_response.members.first).to be_a(Prescription)
+        expect(client_response.cached?).to eq(caching_enabled)
+
+        if caching_enabled
+          expect(cache_key_for(client_response)).to eq("#{client.session.user_id}:gethistoryrx")
+        else
+          expect(cache_key_for(client_response)).to eq(nil)
+        end
+      end
     end
 
-    it 'refills a prescription', :vcr do
-      client_response = client.post_refill_rx(13_650_545)
-      expect(client_response.status).to equal 200
-      # This is what MHV returns, even though we don't care
-      expect(client_response.body).to eq(status: 'success')
+    it 'gets a single prescription' do
+      VCR.use_cassette('rx_client/prescriptions/gets_a_single_prescription') do
+        expect(client.get_rx(13_650_546)).to be_a(Prescription)
+      end
+    end
+
+    it 'refills a prescription' do
+      VCR.use_cassette('rx_client/prescriptions/refills_a_prescription') do
+        if caching_enabled
```

Comment:
did you put this conditional here vs in the `let(:cache_keys)` definition block for any reason in particular? I don't have any feelings about where it goes, just curious.

Reply:
i think I just did that to not have to duplicate specs.